core/src/test/scala/unit/kafka/coordinator/transaction/TransactionStateManagerTest.scala

```diff
@@ -337,6 +337,62 @@ class TransactionStateManagerTest {
     verifyWritesTxnMarkersInPrepareState(PrepareAbort)
   }
 
+  @Test
+  def shouldRemoveExpiredTransactions(): Unit = {
```

Comment:
I don't see how this is possible. You'd have to first go through `initPid` which we will return a retriable exceptions for as the metadata would be in a pending state.